src/ert/config/gen_kw_config.py

```diff
@@ -326,32 +327,19 @@ def write_to_runpath(
         real_nr: int,
         ensemble: Ensemble,
     ) -> dict[str, dict[str, float | str]]:
-        array = ensemble.load_parameters(self.name, real_nr, transformed=True)["values"]
-        assert isinstance(array, xr.DataArray)
-        if not array.size == len(self.transform_functions):
+        df = ensemble.load_parameters(self.name, real_nr, transformed=True).drop(
+            "realization"
+        )
+
+        assert isinstance(df, pl.DataFrame)
+        if not df.width == len(self.transform_functions):
             raise ValueError(
                 f"The configuration of GEN_KW parameter {self.name}"
-                f" is of size {len(self.transform_functions)}, expected {array.size}"
+                f" is of size {len(self.transform_functions)}, expected {df.width}"
```

Comment:
Nitpick: Error message, we expect df.width to correspond to len(self.transform functions), df.width is not the "expected" value.

```suggestion
                f" has {len(self.transform_functions)} parameters, but ensemble dataset for realization {real_nr} has {df.width} parameters."
```